src/OpenDiffix.Core/QueryEngine.fs

```diff
@@ -15,6 +15,32 @@ let rec private extractColumns query =
 
 type QueryResult = { Columns: Column list; Rows: Row list }
 
+// The main entrypoint to the `OpenDiffix.Core` library.
+// Takes in a SQL `statement` as main parameter along with a set of
+// parameters in `queryContext`.
+//
+// Returns the query result as a list of data rows, along with the
+// columns specifications.
+//
+// Supports two main modes of operation:
+// I. Standard SQL statement, with `aidColumns` specified within the `queryContext`
+//
+// Will rewrite the statement so that the query result is anonymized according to the
+// `aidColumns` provided. For example, it will append `AND NOT diffix_low_count(aidColumns)`
+// to the `HAVING` SQL clause and rewrite regural SQL aggregators to Diffix aggregators
+// (e.g. `count` -> `diffix_low_count`, see below)
+//
+// II. SQL statement with Diffix aggregators, no `aidColumns` specified within the `queryContext`
```

Comment:
Can also be without any custom aggregators i.e. regular query.

Reply:
technically true, but then it doesn't make sense to use it, hence not a "main mode of operation". I'll make it clearer that here a "mode of operation that brings the user closer to anonymous results this way or another" is meant.